pkg/agent/datapath/policyBridge.go

```diff
@@ -436,60 +480,93 @@ func (p *PolicyBridge) RemoveLocalEndpoint(endpoint *Endpoint) error {
-		case POLICY_TIER0:
-			policyTable = p.egressTier0PolicyTable
-			nextTable = p.egressTier1PolicyTable
-		case POLICY_TIER1:
-			policyTable = p.egressTier1PolicyTable
-			nextTable = p.ctCommitTable
-		case POLICY_TIER2:
-			policyTable = p.egressTier2PolicyTable
-			nextTable = p.ctCommitTable
-		default:
-			return nil, nil, errors.New("unknow policy tier")
+	switch mode {
+	case "work":
+		switch direction {
+		case POLICY_DIRECTION_OUT:
+			switch tier {
+			case POLICY_TIER1:
+				policyTable = p.egressTier1PolicyTable
+				nextTable = p.egressTier2PolicyTable
+			case POLICY_TIER2:
+				policyTable = p.egressTier2PolicyTable
+				nextTable = p.ctCommitTable
+			case POLICY_TIER3:
+				policyTable = p.egressTier3PolicyTable
+				nextTable = p.ctCommitTable
+			default:
+				return nil, nil, errors.New("unknow policy tier")
+			}
+		case POLICY_DIRECTION_IN:
+			switch tier {
+			case POLICY_TIER1:
+				policyTable = p.ingressTier1PolicyTable
+				nextTable = p.ingressTier2PolicyTable
+			case POLICY_TIER2:
+				policyTable = p.ingressTier2PolicyTable
+				nextTable = p.ctCommitTable
+			case POLICY_TIER3:
+				policyTable = p.ingressTier3PolicyTable
+				nextTable = p.ctCommitTable
+			default:
+				return nil, nil, errors.New("unknow policy tier")
+			}
 		}
-	case POLICY_DIRECTION_IN:
-		switch tier {
-		case POLICY_TIER0:
-			policyTable = p.ingressTier0PolicyTable
-			nextTable = p.ingressTier1PolicyTable
-		case POLICY_TIER1:
-			policyTable = p.ingressTier1PolicyTable
-			nextTable = p.ctCommitTable
-		case POLICY_TIER2:
-			policyTable = p.ingressTier2PolicyTable
-			nextTable = p.ctCommitTable
-		default:
-			return nil, nil, errors.New("unknow policy tier")
+	case "monitor":
+		switch direction {
+		case POLICY_DIRECTION_OUT:
+			switch tier {
+			case POLICY_TIER1:
+			case POLICY_TIER2:
+				policyTable = p.egressTier2PolicyMonitorTable
+				nextTable = p.egressTier2PolicyTable
+			case POLICY_TIER3:
+				policyTable = p.egressTier3PolicyMonitorTable
+				nextTable = p.egressTier3PolicyTable
+			default:
+				return nil, nil, errors.New("unknow policy tier")
+			}
+		case POLICY_DIRECTION_IN:
+			switch tier {
+			case POLICY_TIER1:
+			case POLICY_TIER2:
+				policyTable = p.ingressTier2PolicyMonitorTable
+				nextTable = p.ingressTier2PolicyTable
+			case POLICY_TIER3:
+				policyTable = p.ingressTier3PolicyMonitorTable
+				nextTable = p.ingressTier3PolicyTable
+			default:
+				return nil, nil, errors.New("unknow policy tier")
+			}
 		}
 	}
 
 	return policyTable, nextTable, nil
 }
 
-func (p *PolicyBridge) AddMicroSegmentRule(rule *EveroutePolicyRule, direction uint8, tier uint8) (*FlowEntry, error) {
+func (p *PolicyBridge) AddMicroSegmentRule(rule *EveroutePolicyRule, direction uint8, tier uint8, mode string) (*FlowEntry, error) {
 	var ipDa *net.IP = nil
 	var ipDaMask *net.IP = nil
 	var ipSa *net.IP = nil
 	var ipSaMask *net.IP = nil
 	var err error
+	if mode == "" {
+		mode = DEFAULT_POLICY_ENFORCEMENT_MODE
+	}
```

Comment:
We could set default value in CRD, not compare it here.

Reply:
ok, I get it